governance-aepp/src/utils/contract.js

```diff
@@ -0,0 +1,86 @@
+import settings from './settings';
+import registryInterface from '../assets/contracts/RegistryInterface.aes';
+import pollContractSource from '../assets/contracts/Poll.aes';
+import pollIrisContractSource from '../assets/contracts/Poll_Iris.aes';
+import {toRefs} from "vue";
+import {sdk, wallet} from "@/utils/wallet";
+
+const contract = {
+  registry: null
+};
+
+// TODO adjust ACI usage
+
+contract.init = async () => {
+  const {networkId} = toRefs(wallet)
+
+  contract.registry = await sdk.getContractInstance({
+    source: registryInterface,
+    contractAddress: settings[networkId.value].contractAddress
+  })
+}
+
+contract.delegation = async (address) => {
+  return (await contract.registry.methods.delegatee(address)).decodedResult;
+};
+
+contract.delegations = async (address) => {
+  const delegationsResult = await contract.registry.methods.delegators(address);
+  return Promise.all(Array.from(delegationsResult.decodedResult.entries()).map(async ([delegator, delegatee]) => {
+    const delegateeDelegations = (await contract.registry.methods.delegators(delegator)).decodedResult;
+    return {
+      delegator: delegator,
+      delegatee: delegatee,
+      delegatorAmount: await sdk.getBalance(delegator),
+      includesIndirectDelegations: delegateeDelegations.length !== 0,
+    };
+  }));
+};
+
+contract.polls = async () => {
+  await contract.init();
+  const polls = await contract.registry.methods.polls();
+  return Array.from(polls.decodedResult.entries());
+};
+
+contract.verifyPollContract = async (pollAddress) => {
```

Comment:
Currently this is still the old way of doing this, haven't tried out the new possibilities as exposed with source hash now. Maybe will do this in a secondary step.